hack/e2e-suite/services.sh

```diff
@@ -278,20 +278,26 @@ function wait_for_service_down() {
 #   $5: pod IDs (sorted)
 function verify_from_container() {
   echo "waiting for $1 at $2:$3"
-  results=($(ssh-to-node "${test_node}" "
-      set -e;
-      sudo docker pull gcr.io/google_containers/busybox >/dev/null;
-      sudo docker run gcr.io/google_containers/busybox sh -c '
-          for i in $(seq -s' ' 1 $(($4*3))); do
-            if wget -q -T 3 -O - http://$2:$3; then
-              echo
-            else
-              exit 1
-            fi
-          done
-      '" | sort -r -n | uniq)) \
-      || error "testing $1 VIP from container failed"
-  found_pods=$(sort_args "${results[@]}")
+  for x in 0 1 2 3 4 5 6 7 8 9; do
+    results=($(ssh-to-node "${test_node}" "
+        set -e;
+        sudo docker pull gcr.io/google_containers/busybox >/dev/null;
+        sudo docker run gcr.io/google_containers/busybox sh -c '
+            for i in $(seq -s' ' 1 $(($4*3))); do
+              if wget -q -T 3 -O - http://$2:$3; then
+                echo
+              else
+                exit 1
+              fi
+            done
+        '" | sort -r -n | uniq)) \
+        || error "testing $1 VIP from container failed"
+    found_pods=$(sort_args "${results[@]}")
+    if [[ "${found_pods}" == "$5" ]]; then
+      break
+    fi
+    sleep 5
```

Comment:
Can you please reduce this a bit once you've confirmed that it addresses the issue?  50 seconds is a long time to wait for something that ain't gonna happen :-)
It would be useful to log how long it took to pass, so that we can tune the timeout accordingly.